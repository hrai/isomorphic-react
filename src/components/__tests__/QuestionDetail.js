import React from 'react';
import renderer from 'react-test-renderer';
import { mapStateToProps, QuestionDetailDisplay } from '../QuestionDetail';

describe(`the question detail component`, () => {
    it("should map the state to props correctly", () => {
        const sampleQuestion = {
            question_id:42,
            body:'First question',
        };

        const appState = {
            questions:[sampleQuestion],
        };
        const ownProps = {
            question_id: 42
        };
        const componentState = mapStateToProps(appState, ownProps);

        expect(componentState).toEqual(sampleQuestion);
    });

    describe('The display element', () => {
        it('Should not regress', () => {
            const tree = renderer.create(
                <QuestionDetailDisplay
                    title='The testament'
                    body='23'
                    answer_count={0}
                    tags={ [ 'hiking' ] }
                />

            );

            expect(tree.toJSON()).toMatchSnapshot();
        });
    });
});
