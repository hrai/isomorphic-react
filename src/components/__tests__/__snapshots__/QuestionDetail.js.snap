// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the question detail component The display element Should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    The testament
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          hiking
        </code>
      </div>
    </div>
    <div>
      <p>
        23
      </p>
    </div>
    <div>
      0
       Answers
    </div>
  </div>
</div>
`;
